common/internal/queues/priority_heap.go

```diff
@@ -0,0 +1,92 @@
+package queues
+
+import (
+	feetypes "github.com/smartcontractkit/chainlink/v2/common/fee/types"
+	txmgrtypes "github.com/smartcontractkit/chainlink/v2/common/txmgr/types"
+	"github.com/smartcontractkit/chainlink/v2/common/types"
+)
+
+// priorityHeap is a priority queue of transactions prioritized by creation time. The oldest transaction is at the front of the queue.
+// It implements the heap interface in the container/heap package and is safe for concurrent access.
```

Comment:
Re: concurrency safe, what if `Swap` and `Pop` are called concurrently?

Reply:
I don't understand. To me it looks like you've taken something that is not concurrent safe `priorityHeap` and made it concurrent safe in `TxPriorityQueue` by adding a locking mechanism (the RWMutex).